### PR TITLE
[build-utils][cli] Add env vars for `VERCEL_PROJECT_SETTINGS_`

### PR DESCRIPTION
**Pull request by styfle**
This PR adds a few new environment variables for project settings.

This allows frameworks targeting the [Build Output API](https://vercel.com/docs/build-output-api/v3) to read data that is normally only available in through the `config` object when developing a [Builder/Runtime](https://github.com/vercel/vercel/blob/main/DEVELOPING_A_RUNTIME.md).

This will also solve the problem of old Builders/Runtimes that never passed the `config` through to `getNodeVersion()`.

- Related to https://github.com/unjs/nitro/pull/879

### CODE REVIEW

**Review by cb1kenobi — DISMISSED** [review]
LGTM

**Review by EndangeredMassa — DISMISSED** [review]

File: packages/cli/test/unit/commands/build/index.test.ts [hunk 1]
```diff
@@ -1025,6 +1025,36 @@ describe('build', () => {
     }
   });
 
+  it('should set VERCEL_PROJECT_SETTINGS_ environment variables', async () => {
+    if (process.platform === 'win32') {
+      // this test runs a build command with `mkdir -p` which is unsupported on Windows
```
Comment (TooTallNate) [inline packages/cli/test/unit/commands/build/index.test.ts]: Convert to a Node.js based build script instead?

File: packages/cli/test/unit/commands/build/index.test.ts [hunk 2]
```diff
@@ -1025,6 +1025,36 @@ describe('build', () => {
...
+    }
+
+    const cwd = fixture('project-settings-env-vars');
+    const output = join(cwd, '.vercel/output');
+    try {
+      process.chdir(cwd);
+      const exitCode = await build(client);
+      expect(exitCode).toEqual(0);
+
+      const contents = await fs.readFile(
```
Comment (TooTallNate) [inline packages/cli/test/unit/commands/build/index.test.ts]: `fs.readJson()`?

**Review by TooTallNate — DISMISSED** [review]
Couple minor non-blocking nits. Looks good!

File: packages/cli/test/unit/commands/build/index.test.ts (hunk 1 above)
Comment (styfle) [reply packages/cli/test/unit/commands/build/index.test.ts]: Done in 6b968c20bafe37cadfb7c9af0800cde53a1e8630

File: packages/cli/test/unit/commands/build/index.test.ts (hunk 2 above)
Comment (styfle) [reply packages/cli/test/unit/commands/build/index.test.ts]: Done in 6b968c20bafe37cadfb7c9af0800cde53a1e8630

**Review by cb1kenobi — DISMISSED** [review]

**Review by TooTallNate — DISMISSED** [review]

File: packages/cli/test/unit/commands/build/index.test.ts [hunk 3]
```diff
@@ -1025,6 +1025,27 @@ describe('build', () => {
...
+  it('should set VERCEL_PROJECT_SETTINGS_ environment variables', async () => {
+    const cwd = fixture('project-settings-env-vars');
+    const output = join(cwd, '.vercel/output');
+    try {
+      process.chdir(cwd);
+      const exitCode = await build(client);
+      expect(exitCode).toEqual(0);
+
+      const contents = await fs.readJSON(join(output, 'static/env.json'));
+      expect(JSON.parse(contents)).toMatchObject({
```
Comment (TooTallNate) [inline packages/cli/test/unit/commands/build/index.test.ts]:
```suggestion
      expect(contents).toMatchObject({
```

**Review by cb1kenobi — APPROVED** [review]

**Pull request merged**